lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -521,14 +521,87 @@ public function getParamType(ParameterReflection $param)
                 $param->getDeclaringFunction()->name
             ));
         }
-        if ($type == 'array') {
+        if ($type === 'array') {
             // try to determine class, if it's array of objects
             $paramDocBlock = $this->getParamDocBlockTag($param);
             $paramTypes = $paramDocBlock->getTypes();
             $paramType = array_shift($paramTypes);
+
+            $paramType = $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $paramType);
+
             return strpos($paramType, '[]') !== false ? $paramType : "{$paramType}[]";
         }
-        return $type;
+
+        return $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $type);
+    }
+
+    /**
+     * Resolve fully qualified type name in the class alias context
+     *
+     * @param ClassReflection $sourceClass
+     * @param string $typeName
+     * @return string
+     */
+    public function resolveFullyQualifiedClassName(ClassReflection $sourceClass, string $typeName): string
+    {
+        $typeName = trim($typeName);
+
+        // Not a class, but a basic type
+        if ((strtolower($typeName[0])) === $typeName[0]) {
+            return $typeName;
+        }
+
+        // Split array suffix
+        if (preg_match('/^(.+?)\[\]$/', $typeName, $matches)) {
+            $typeName = $matches[1];
+            $arraySuffix = '[]';
+        } else {
+            $arraySuffix = '';
+        }
+
+        // If class exists, then return it
+        try {
+            new ClassReflection($typeName);
```

Comment:
Is it possible to verify that class exists in with a different approach? I would consider exception swallowing in `catch` block as a sign to refactor this part